tt_metal/hw/inc/dataflow_api.h

```diff
@@ -1551,40 +1631,79 @@ class Buffer {
             case BufferType::SYSTEM_MEMORY: this->get_noc_addr_helper = get_system_memory_noc_addr; break;
         }
     }
-    uint64_t get_noc_addr(const uint32_t id, const uint32_t offset = 0) {
+    uint64_t get_noc_addr_(const uint32_t id, const uint32_t offset = 0) {
+        uint64_t noc_addr = this->get_noc_addr_helper(id, this->page_size_, this->bank_base_address, offset);
         return this->get_noc_addr_helper(id, this->page_size_, this->bank_base_address, offset);
     }
 
    public:
 
+    Buffer(){;}
+
     Buffer(const BufferType type, const uint32_t bank_base_address, const uint32_t page_size) {
+        this->init(type, bank_base_address, page_size);
+    }
+
+    Buffer(uint32_t page_size, uint32_t num_cores,  uint32_t addr, volatile tt_l1_ptr uint32_t* command_ptr) {
+        this->init_sharded(page_size, num_cores, addr, command_ptr);
+    }
+
+    void init(const BufferType type, const uint32_t bank_base_address, const uint32_t page_size) {
         this->set_type(type);
         this->bank_base_address = bank_base_address;
         this->page_size_ = page_size;
+        this->sharded = false;
     }
 
+    void init_sharded(uint32_t page_size, uint32_t num_cores,  uint32_t addr, volatile tt_l1_ptr uint32_t* command_ptr){
```

Comment:
would users want/need to call these functions again after `Buffer` is constructed?